hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java

```diff
@@ -122,4 +132,25 @@ public void handle(@NonNull final HandleContext context) throws HandleException
         It will not be committed to state until commit is called on the state.--- */
         fileStore.put(fileBuilder.build());
     }
+
+    private static void handleAppendUpgradeFile(FileAppendTransactionBody fileAppend, HandleContext handleContext) {
+        final var fileStore = handleContext.writableStore(WritableUpgradeStore.class);
+        File file = fileStore.peek();
+        if (file == null) {
+            throw new HandleException(INVALID_FILE_ID);
+        }
+        final var appendedFile = new File.Builder()
+                // copy over just the fileNum from the update file
+                // leave shard and realm at zero
+                // this ensures that shard and realm are clamped to zero
+                .fileId(FileID.newBuilder()
+                        .fileNum(fileAppend.fileIDOrThrow().fileNum())
+                        .build())
+                .contents(fileAppend.contents())
+                .deleted(false)
+                .expirationTime(file.expirationTime())
+                .memo(file.memo())
+                .build();
+        fileStore.add(appendedFile);
```

Comment:
_Nit_ - The_`FileAppend` semantics affect _only_ the file's content, not its metadata.

So I think there should be a `fileStore.append(Bytes extraContents)` method we use here; not an `add()` that takes an entire `File` object.

Reply:
The file add is doing excatly it inside.
```
    public void add(@NonNull File file) {
        requireNonNull(file);
        writableUpgradeState.add(file.contents());
        writableUpgradeFileState.put(file);
    }
```

Reply:
I fixed it as you asked for the appendHandler inside the store